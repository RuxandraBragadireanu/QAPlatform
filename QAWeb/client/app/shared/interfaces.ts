export interface Topic {
  id: number,
  title: string,
  userId: string,
  userName: string,
  content: string,
  comments: Comment[]
}

export interface Comment {
  id: number,
  content: string,
  score: number,
  user: {
    id: number,
    username: string
  }
}

export interface MatchProps {
  params: any,
  isExact: boolean,
  path: string,
  url: string
}

