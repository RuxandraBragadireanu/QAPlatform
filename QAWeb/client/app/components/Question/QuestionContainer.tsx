import * as React from 'react';
import {connect} from "react-redux";
import {QuestionActions} from "./store/question.actions";
import { ApplicationState } from '../../store/application-state';
import { MatchProps, Topic } from '../../shared/interfaces';
import { Button, CircularProgress, Paper, TableCell } from '@material-ui/core';
import { ValidatorForm, TextValidator} from 'react-material-ui-form-validator';
import DeleteForeverIcon from '@material-ui/icons/DeleteForever';

interface TopicProps {
  topic: Topic,
  isLoading: boolean,
  match: MatchProps,
  onLoad: Function,
  addComment: Function,
  deleteComment: Function,
}

interface TopicState {
  comment: string
}

export class QuestionContainer extends React.Component<TopicProps, TopicState> {

  constructor(props) {
    super(props);

    this.state = {
      comment: ''
    }
  }

  componentWillMount() {
    this.props.onLoad(this.props.match.params.id);
  }

  addComment = () => {
    this.props.addComment(this.props.topic.id, this.state.comment);
  };

  changeComment = (event) => {
    this.setState({comment: event.target.value})
  };

  deleteComment = (commentId) => {
    this.props.deleteComment(commentId);
  };

  componentWillReceiveProps() {
    this.setState({comment: ''});
  }

  render() {
    const { topic } = this.props;

    const comments = topic.comments && topic.comments.map(comment => (
      <Paper key={comment.id} style={{width: 'calc(54% + 10px)', margin: 24, marginLeft: '23%', overflowX: 'auto'}}>
        <div style={{padding: 24, wordBreak: 'break-word'}}>
          <b>{comment.user.username}</b>
            <div style={{display: 'flex'}}>
              <div style={{marginTop: 8}}>
                {comment.content}
              </div>
              <div onClick={() => this.deleteComment(comment.id)} style={{cursor: 'pointer', marginLeft: '54%', marginTop: -8, position: 'absolute', display: 'flex'}}>
                <DeleteForeverIcon/>
              </div>
            </div>
        </div>
      </Paper>
    ));

    return(
      <div style={{marginBottom: '10%'}}>
        <div style={{marginLeft: '23%'}}>
        <h1>{topic && topic.title}</h1>

          {this.props.topic.content && (<Paper style={{width: 'calc(70% - 10px)', overflowX: 'auto', marginTop: 16, padding: 10}}>
            <div>
              {this.props.topic.content}
            </div>
          </Paper>)}


        <ValidatorForm onSubmit={(event) => {
          event.preventDefault();
          return this.addComment()
        }}>

          <TextValidator
            id="standard-textarea"
            label="Comment"
            name="comment"
            value={this.state.comment}
            onChange={this.changeComment}
            style={{width: 'calc(70% + 12px)'}}
            margin="normal"
            InputLabelProps={{
              shrink: true
            }}
            validators={['required']}
            errorMessages={['This field is required']}
            variant="outlined"
            multiline
          />

          <div>
            <Button type="submit" variant='contained'>
              Submit
            </Button>
          </div>

        </ValidatorForm>

        </div>

        <div>
          {comments}

          {this.props.isLoading && <CircularProgress style={{marginLeft: '48%', marginTop: 24}}/>}
          </div>
      </div>
    )
  }
}

const mapStateToProps = (state : ApplicationState) => {
  return {
    topic: state.topic.topic,
    isLoading: state.topic.isLoading
  }
};

const mapDispatchToProps = dispatch => {
  return {
    onLoad: (id) => {
      dispatch(QuestionActions.loadTopicStart(id));
    },
    addComment: (topicId, comment) => {
      dispatch(QuestionActions.addCommentStart(topicId, comment));
    },
    deleteComment: (commentId) => {
      dispatch(QuestionActions.deleteComment(commentId));
    }
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(QuestionContainer);
