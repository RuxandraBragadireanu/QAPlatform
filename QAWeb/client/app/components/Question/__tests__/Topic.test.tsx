import { QuestionContainer } from '../QuestionContainer';
import * as React from 'react';
import { shallow, mount, render } from 'enzyme';
import { Topic } from '../../../shared/interfaces';


describe('Topic', () => {
});
