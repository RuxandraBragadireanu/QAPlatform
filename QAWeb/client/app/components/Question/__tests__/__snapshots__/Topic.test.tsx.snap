// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic should render correctly, and match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionContainer
    addComment={[Function]}
    deleteComment={[Function]}
    isLoading={false}
    match={
      Object {
        "params": Object {
          "id": 100,
        },
      }
    }
    onLoad={[Function]}
    topic={
      Object {
        "comments": Array [
          Object {
            "description": "TestDescription",
            "id": 100,
            "user": Object {
              "id": 1000,
              "username": "TestUserName2",
            },
          },
        ],
        "id": 1,
        "title": "TestTitle1",
        "userName": "TestUsername1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          style={
            Object {
              "marginLeft": 24,
            }
          }
        >
          <h1>
            TestTitle1
          </h1>
          <ValidatorForm
            debounceTime={0}
            onError={[Function]}
            onSubmit={[Function]}
          >
            <TextValidator
              InputLabelProps={
                Object {
                  "shrink": true,
                }
              }
              errorMessages={
                Array [
                  "This field is required",
                ]
              }
              id="standard-textarea"
              label="Comment"
              margin="normal"
              multiline={true}
              name="comment"
              onChange={[Function]}
              style={
                Object {
                  "width": "calc(40% + 12px)",
                }
              }
              validatorListener={[Function]}
              validators={
                Array [
                  "required",
                ]
              }
              value=""
              variant="outlined"
            />
            <div>
              <WithStyles(Button)
                type="submit"
                variant="contained"
              >
                Submit
              </WithStyles(Button)>
            </div>
          </ValidatorForm>
        </div>,
        <div>
          <WithStyles(Paper)
            style={
              Object {
                "margin": 24,
                "overflowX": "auto",
                "width": "40%",
              }
            }
          >
            <div
              style={
                Object {
                  "padding": 24,
                  "wordBreak": "break-word",
                }
              }
            >
              <b>
                TestUserName2
              </b>
              <div
                style={
                  Object {
                    "display": "flex",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "marginTop": 8,
                    }
                  }
                >
                  TestDescription
                </div>
                <div
                  onClick={[Function]}
                  style={
                    Object {
                      "cursor": "pointer",
                      "display": "flex",
                      "marginLeft": "40%",
                      "marginTop": -8,
                      "position": "absolute",
                    }
                  }
                >
                  <pure(DeleteForeverIcon) />
                </div>
              </div>
            </div>
          </WithStyles(Paper)>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              TestTitle1
            </h1>,
            <ValidatorForm
              debounceTime={0}
              onError={[Function]}
              onSubmit={[Function]}
            >
              <TextValidator
                InputLabelProps={
                  Object {
                    "shrink": true,
                  }
                }
                errorMessages={
                  Array [
                    "This field is required",
                  ]
                }
                id="standard-textarea"
                label="Comment"
                margin="normal"
                multiline={true}
                name="comment"
                onChange={[Function]}
                style={
                  Object {
                    "width": "calc(40% + 12px)",
                  }
                }
                validatorListener={[Function]}
                validators={
                  Array [
                    "required",
                  ]
                }
                value=""
                variant="outlined"
              />
              <div>
                <WithStyles(Button)
                  type="submit"
                  variant="contained"
                >
                  Submit
                </WithStyles(Button)>
              </div>
            </ValidatorForm>,
          ],
          "style": Object {
            "marginLeft": 24,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "TestTitle1",
            },
            "ref": null,
            "rendered": "TestTitle1",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <TextValidator
                  InputLabelProps={
                    Object {
                      "shrink": true,
                    }
                  }
                  errorMessages={
                    Array [
                      "This field is required",
                    ]
                  }
                  id="standard-textarea"
                  label="Comment"
                  margin="normal"
                  multiline={true}
                  name="comment"
                  onChange={[Function]}
                  style={
                    Object {
                      "width": "calc(40% + 12px)",
                    }
                  }
                  validatorListener={[Function]}
                  validators={
                    Array [
                      "required",
                    ]
                  }
                  value=""
                  variant="outlined"
                />,
                <div>
                  <WithStyles(Button)
                    type="submit"
                    variant="contained"
                  >
                    Submit
                  </WithStyles(Button)>
                </div>,
              ],
              "debounceTime": 0,
              "onError": [Function],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "InputLabelProps": Object {
                    "shrink": true,
                  },
                  "errorMessages": Array [
                    "This field is required",
                  ],
                  "id": "standard-textarea",
                  "label": "Comment",
                  "margin": "normal",
                  "multiline": true,
                  "name": "comment",
                  "onChange": [Function],
                  "style": Object {
                    "width": "calc(40% + 12px)",
                  },
                  "validatorListener": [Function],
                  "validators": Array [
                    "required",
                  ],
                  "value": "",
                  "variant": "outlined",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <WithStyles(Button)
                    type="submit"
                    variant="contained"
                  >
                    Submit
                  </WithStyles(Button)>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Submit",
                    "type": "submit",
                    "variant": "contained",
                  },
                  "ref": null,
                  "rendered": "Submit",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            Array [
              <WithStyles(Paper)
                style={
                  Object {
                    "margin": 24,
                    "overflowX": "auto",
                    "width": "40%",
                  }
                }
              >
                <div
                  style={
                    Object {
                      "padding": 24,
                      "wordBreak": "break-word",
                    }
                  }
                >
                  <b>
                    TestUserName2
                  </b>
                  <div
                    style={
                      Object {
                        "display": "flex",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "marginTop": 8,
                        }
                      }
                    >
                      TestDescription
                    </div>
                    <div
                      onClick={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                          "display": "flex",
                          "marginLeft": "40%",
                          "marginTop": -8,
                          "position": "absolute",
                        }
                      }
                    >
                      <pure(DeleteForeverIcon) />
                    </div>
                  </div>
                </div>
              </WithStyles(Paper)>,
            ],
            false,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "100",
            "nodeType": "class",
            "props": Object {
              "children": <div
                style={
                  Object {
                    "padding": 24,
                    "wordBreak": "break-word",
                  }
                }
              >
                <b>
                  TestUserName2
                </b>
                <div
                  style={
                    Object {
                      "display": "flex",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "marginTop": 8,
                      }
                    }
                  >
                    TestDescription
                  </div>
                  <div
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "flex",
                        "marginLeft": "40%",
                        "marginTop": -8,
                        "position": "absolute",
                      }
                    }
                  >
                    <pure(DeleteForeverIcon) />
                  </div>
                </div>
              </div>,
              "style": Object {
                "margin": 24,
                "overflowX": "auto",
                "width": "40%",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    TestUserName2
                  </b>,
                  <div
                    style={
                      Object {
                        "display": "flex",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "marginTop": 8,
                        }
                      }
                    >
                      TestDescription
                    </div>
                    <div
                      onClick={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                          "display": "flex",
                          "marginLeft": "40%",
                          "marginTop": -8,
                          "position": "absolute",
                        }
                      }
                    >
                      <pure(DeleteForeverIcon) />
                    </div>
                  </div>,
                ],
                "style": Object {
                  "padding": 24,
                  "wordBreak": "break-word",
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TestUserName2",
                  },
                  "ref": null,
                  "rendered": "TestUserName2",
                  "type": "b",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <div
                        style={
                          Object {
                            "marginTop": 8,
                          }
                        }
                      >
                        TestDescription
                      </div>,
                      <div
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "flex",
                            "marginLeft": "40%",
                            "marginTop": -8,
                            "position": "absolute",
                          }
                        }
                      >
                        <pure(DeleteForeverIcon) />
                      </div>,
                    ],
                    "style": Object {
                      "display": "flex",
                    },
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "TestDescription",
                        "style": Object {
                          "marginTop": 8,
                        },
                      },
                      "ref": null,
                      "rendered": "TestDescription",
                      "type": "div",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": <pure(DeleteForeverIcon) />,
                        "onClick": [Function],
                        "style": Object {
                          "cursor": "pointer",
                          "display": "flex",
                          "marginLeft": "40%",
                          "marginTop": -8,
                          "position": "absolute",
                        },
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "class",
                        "props": Object {},
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      "type": "div",
                    },
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": [Function],
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            style={
              Object {
                "marginLeft": 24,
              }
            }
          >
            <h1>
              TestTitle1
            </h1>
            <ValidatorForm
              debounceTime={0}
              onError={[Function]}
              onSubmit={[Function]}
            >
              <TextValidator
                InputLabelProps={
                  Object {
                    "shrink": true,
                  }
                }
                errorMessages={
                  Array [
                    "This field is required",
                  ]
                }
                id="standard-textarea"
                label="Comment"
                margin="normal"
                multiline={true}
                name="comment"
                onChange={[Function]}
                style={
                  Object {
                    "width": "calc(40% + 12px)",
                  }
                }
                validatorListener={[Function]}
                validators={
                  Array [
                    "required",
                  ]
                }
                value=""
                variant="outlined"
              />
              <div>
                <WithStyles(Button)
                  type="submit"
                  variant="contained"
                >
                  Submit
                </WithStyles(Button)>
              </div>
            </ValidatorForm>
          </div>,
          <div>
            <WithStyles(Paper)
              style={
                Object {
                  "margin": 24,
                  "overflowX": "auto",
                  "width": "40%",
                }
              }
            >
              <div
                style={
                  Object {
                    "padding": 24,
                    "wordBreak": "break-word",
                  }
                }
              >
                <b>
                  TestUserName2
                </b>
                <div
                  style={
                    Object {
                      "display": "flex",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "marginTop": 8,
                      }
                    }
                  >
                    TestDescription
                  </div>
                  <div
                    onClick={[Function]}
                    style={
                      Object {
                        "cursor": "pointer",
                        "display": "flex",
                        "marginLeft": "40%",
                        "marginTop": -8,
                        "position": "absolute",
                      }
                    }
                  >
                    <pure(DeleteForeverIcon) />
                  </div>
                </div>
              </div>
            </WithStyles(Paper)>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                TestTitle1
              </h1>,
              <ValidatorForm
                debounceTime={0}
                onError={[Function]}
                onSubmit={[Function]}
              >
                <TextValidator
                  InputLabelProps={
                    Object {
                      "shrink": true,
                    }
                  }
                  errorMessages={
                    Array [
                      "This field is required",
                    ]
                  }
                  id="standard-textarea"
                  label="Comment"
                  margin="normal"
                  multiline={true}
                  name="comment"
                  onChange={[Function]}
                  style={
                    Object {
                      "width": "calc(40% + 12px)",
                    }
                  }
                  validatorListener={[Function]}
                  validators={
                    Array [
                      "required",
                    ]
                  }
                  value=""
                  variant="outlined"
                />
                <div>
                  <WithStyles(Button)
                    type="submit"
                    variant="contained"
                  >
                    Submit
                  </WithStyles(Button)>
                </div>
              </ValidatorForm>,
            ],
            "style": Object {
              "marginLeft": 24,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "TestTitle1",
              },
              "ref": null,
              "rendered": "TestTitle1",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <TextValidator
                    InputLabelProps={
                      Object {
                        "shrink": true,
                      }
                    }
                    errorMessages={
                      Array [
                        "This field is required",
                      ]
                    }
                    id="standard-textarea"
                    label="Comment"
                    margin="normal"
                    multiline={true}
                    name="comment"
                    onChange={[Function]}
                    style={
                      Object {
                        "width": "calc(40% + 12px)",
                      }
                    }
                    validatorListener={[Function]}
                    validators={
                      Array [
                        "required",
                      ]
                    }
                    value=""
                    variant="outlined"
                  />,
                  <div>
                    <WithStyles(Button)
                      type="submit"
                      variant="contained"
                    >
                      Submit
                    </WithStyles(Button)>
                  </div>,
                ],
                "debounceTime": 0,
                "onError": [Function],
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "InputLabelProps": Object {
                      "shrink": true,
                    },
                    "errorMessages": Array [
                      "This field is required",
                    ],
                    "id": "standard-textarea",
                    "label": "Comment",
                    "margin": "normal",
                    "multiline": true,
                    "name": "comment",
                    "onChange": [Function],
                    "style": Object {
                      "width": "calc(40% + 12px)",
                    },
                    "validatorListener": [Function],
                    "validators": Array [
                      "required",
                    ],
                    "value": "",
                    "variant": "outlined",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <WithStyles(Button)
                      type="submit"
                      variant="contained"
                    >
                      Submit
                    </WithStyles(Button)>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Submit",
                      "type": "submit",
                      "variant": "contained",
                    },
                    "ref": null,
                    "rendered": "Submit",
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              Array [
                <WithStyles(Paper)
                  style={
                    Object {
                      "margin": 24,
                      "overflowX": "auto",
                      "width": "40%",
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "padding": 24,
                        "wordBreak": "break-word",
                      }
                    }
                  >
                    <b>
                      TestUserName2
                    </b>
                    <div
                      style={
                        Object {
                          "display": "flex",
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "marginTop": 8,
                          }
                        }
                      >
                        TestDescription
                      </div>
                      <div
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "flex",
                            "marginLeft": "40%",
                            "marginTop": -8,
                            "position": "absolute",
                          }
                        }
                      >
                        <pure(DeleteForeverIcon) />
                      </div>
                    </div>
                  </div>
                </WithStyles(Paper)>,
              ],
              false,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "100",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  style={
                    Object {
                      "padding": 24,
                      "wordBreak": "break-word",
                    }
                  }
                >
                  <b>
                    TestUserName2
                  </b>
                  <div
                    style={
                      Object {
                        "display": "flex",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "marginTop": 8,
                        }
                      }
                    >
                      TestDescription
                    </div>
                    <div
                      onClick={[Function]}
                      style={
                        Object {
                          "cursor": "pointer",
                          "display": "flex",
                          "marginLeft": "40%",
                          "marginTop": -8,
                          "position": "absolute",
                        }
                      }
                    >
                      <pure(DeleteForeverIcon) />
                    </div>
                  </div>
                </div>,
                "style": Object {
                  "margin": 24,
                  "overflowX": "auto",
                  "width": "40%",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      TestUserName2
                    </b>,
                    <div
                      style={
                        Object {
                          "display": "flex",
                        }
                      }
                    >
                      <div
                        style={
                          Object {
                            "marginTop": 8,
                          }
                        }
                      >
                        TestDescription
                      </div>
                      <div
                        onClick={[Function]}
                        style={
                          Object {
                            "cursor": "pointer",
                            "display": "flex",
                            "marginLeft": "40%",
                            "marginTop": -8,
                            "position": "absolute",
                          }
                        }
                      >
                        <pure(DeleteForeverIcon) />
                      </div>
                    </div>,
                  ],
                  "style": Object {
                    "padding": 24,
                    "wordBreak": "break-word",
                  },
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TestUserName2",
                    },
                    "ref": null,
                    "rendered": "TestUserName2",
                    "type": "b",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <div
                          style={
                            Object {
                              "marginTop": 8,
                            }
                          }
                        >
                          TestDescription
                        </div>,
                        <div
                          onClick={[Function]}
                          style={
                            Object {
                              "cursor": "pointer",
                              "display": "flex",
                              "marginLeft": "40%",
                              "marginTop": -8,
                              "position": "absolute",
                            }
                          }
                        >
                          <pure(DeleteForeverIcon) />
                        </div>,
                      ],
                      "style": Object {
                        "display": "flex",
                      },
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "TestDescription",
                          "style": Object {
                            "marginTop": 8,
                          },
                        },
                        "ref": null,
                        "rendered": "TestDescription",
                        "type": "div",
                      },
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": <pure(DeleteForeverIcon) />,
                          "onClick": [Function],
                          "style": Object {
                            "cursor": "pointer",
                            "display": "flex",
                            "marginLeft": "40%",
                            "marginTop": -8,
                            "position": "absolute",
                          },
                        },
                        "ref": null,
                        "rendered": Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "class",
                          "props": Object {},
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        "type": "div",
                      },
                    ],
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
